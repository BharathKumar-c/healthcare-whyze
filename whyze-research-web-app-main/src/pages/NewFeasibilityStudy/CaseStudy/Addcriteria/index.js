import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import { Card } from 'antd';
import { Segaments } from '../../../../components/basic';
import { Modal } from '../../../../components/hoc';
import { IconList } from '../../../../constant/IconData';
import './AddCriteria.scss';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import {
  setCardDetails,
  setSegments,
} from '../../../../redux/reducer/feasibilityStudyReducer';
import {
  addCriteriaTitle,
  cancelButtonLabel,
  saveButtonLabel,
} from '../../../../constant/ConstantTexts';

export default function AddCriteria(props) {
  const [titles, setTitles] = useState([]);
  const [isloading, setIsLoading] = useState(false);
  const [segmentoption, setSegmentOption] = useState('Inclusion');
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();

  const Titles = (ele) => {
    const newTitleChange = [...titles];
    const findIndex = newTitleChange?.findIndex(
      (element) => element === ele.Title,
    );

    if (findIndex > -1) {
      newTitleChange?.splice(findIndex, 1);
    } else {
      newTitleChange?.push(ele.Title);
    }
    setTitles(newTitleChange);
  };

  const SaveConditions = (titles) => {
    setIsLoading(true);
    const caseStudy = JSON.parse(JSON.stringify(segments));
    titles.map((elem) => {
      caseStudy[activeSegment.index].feasibilityStudy.push({
        name: elem,
        case: elem,
        condition: segmentoption,
        caseName: '',
        child:
          elem === 'Demographics'
            ? [
                {
                  id: uuid().slice(0, 8),
                  condition: 'Gender',
                  field: 'gender',
                  gender: '',
                },
                {
                  id: uuid().slice(0, 8),
                  condition: 'Age range',
                  field: 'age',
                  ageRange: {
                    min: 0,
                    max: 0,
                  },
                },
              ]
            : elem === 'Weight'
            ? [
                {
                  id: uuid().slice(0, 8),
                  condition: 'Weight range',
                  field: 'weight',
                  weightRange: {
                    min: 0,
                    max: 0,
                    range: '',
                  },
                },
              ]
            : elem === 'BMI'
            ? [
                {
                  id: uuid().slice(0, 8),
                  condition: 'Bmi range',
                  bmiRange: '',
                },
              ]
            : [],
      });
    });

    dispatch(
      setCardDetails({
        index: caseStudy[activeSegment.index].feasibilityStudy.length - 1,
        case: titles[titles.length - 1],
        condition: segmentoption,
      }),
    );

    dispatch(setSegments(caseStudy));
    setTitles([]);
    setIsLoading(false);
    props.onClose(false);
  };

  const handleClose = () => {
    setTitles([]);
    props.onClose(false);
  };

  return (
    <div className="AddCriteria-container">
      <Modal
        isOpen={props.isOpen}
        onClose={handleClose}
        closable
        title={addCriteriaTitle}
        width={1000}
        handleActionClick={() => SaveConditions(titles)}
        okButtonLabel={saveButtonLabel}
        cancelButtonLabel={cancelButtonLabel}
        confirmLoading={isloading}
      >
        <Segaments
          options={['Inclusion', 'Exclusion']}
          size={'large'}
          onChange={(e) => setSegmentOption(e)}
        />
        <div className="AddCriteria-container_Card-wrapper">
          {IconList.map((ele, index) => {
            if (ele?.Title !== 'Location') {
              return (
                <Card
                  data-testid="case-study-card"
                  key={index}
                  style={{
                    borderRadius: '10px',
                    border: '1px solid  rgba(45, 51, 57, 0.12)',
                    cursor: 'pointer',
                    backgroundColor: titles.includes(ele.Title)
                      ? '#F6F7F9'
                      : 'white',
                  }}
                  onClick={() => Titles(ele)}
                >
                  <div className="AddCriteria-container_card-elements">
                    <div>{ele.Icons}</div>
                    <div>{ele.Title}</div>
                  </div>
                </Card>
              );
            }
          })}
        </div>
      </Modal>
    </div>
  );
}

AddCriteria.propTypes = {
  isOpen: PropTypes.bool,
  onClose: PropTypes.func,
};
AddCriteria.defaultProps = {
  isOpen: false,
  onClose: () => {},
};
