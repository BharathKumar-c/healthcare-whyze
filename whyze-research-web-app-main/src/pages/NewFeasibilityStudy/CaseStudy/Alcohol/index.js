import React from 'react';
import { Image, Input } from '../../../../components/basic';
import AddIcon from '../../../../assets/AddIcon.svg';
import PropTypes from 'prop-types';

const Alcohol = () => {
  return (
    <div>
      <Input id="main_textbox" lable="Alcohol condition" />
      <div style={{ height: '30px' }}>
        <Image
          style={{ cursor: 'pointer' }}
          preview={false}
          width={12}
          height={12}
          src={AddIcon}
        />
      </div>
    </div>
  );
};

Alcohol.propTypes = {
  caseDetails: PropTypes.array,
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Alcohol.defaultProps = {
  caseDetails: [],
  parentIndex: 0,
  isChild: false,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};
export default Alcohol;
