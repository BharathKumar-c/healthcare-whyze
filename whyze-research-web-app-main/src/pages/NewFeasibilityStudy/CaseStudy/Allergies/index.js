import React, { useEffect, useState } from 'react';
import { Image, AutoCompleteComponent } from '../../../../components/basic';
import AddIcon from '../../../../assets/AddIcon.svg';
import PropTypes from 'prop-types';
import {
  AllergiesOptions,
  Allergiesarray,
  autocompleteDefaultLabel,
  autocompleteEmptyLabel,
  Symptomarray,
} from '../../../../constant';

import AddCondition from '../../../../components/shared/AddCondition';
import Reactions from './Reactions';
import useNewFeasibilityStudyService from '../../useNewFeasibilityStudyService';
import { useSelector } from 'react-redux';
import { allergiesConstants } from '../../../../constant/ConstantTexts';

const Allergies = (props) => {
  const [isError, setIsError] = useState(false);
  const [isGetCount, setIsGetCount] = useState(false);
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const { optimusValue } = useNewFeasibilityStudyService();

  const handleError = () => {
    setIsError(true);
  };
  const handleChangeCaseName = (e) => {
    props.handleChangeCaseName(e, props.parentIndex);
    setIsError(false);
  };

  const handleSelectCaseName = (e) => {
    props.handleChangeCaseName(e, props.parentIndex);
    setIsError(false);
    setIsGetCount(true);
  };

  useEffect(() => {
    if (isGetCount) {
      optimusValue({
        name: props?.caseDetails?.name,
        data: segments[activeSegment.index]?.feasibilityStudy,
      });
      setIsGetCount(false);
    }
  }, [isGetCount]);

  return (
    <div className="indication" id="indication">
      {!props?.isChild && (
        <div>
          <AutoCompleteComponent
            options={
              props?.caseDetails?.caseName?.length >= 3
                ? Allergiesarray.length > 0
                  ? Allergiesarray
                  : autocompleteEmptyLabel
                : autocompleteDefaultLabel
            }
            getPopupContainer={() => document.getElementById('indication')}
            value={props?.caseDetails?.caseName}
            placeholder={allergiesConstants.allergiesPlaceholder}
            label={allergiesConstants.allergiesLabel}
            handleSearch={handleChangeCaseName}
            handleSelect={handleSelectCaseName}
            errorText={allergiesConstants.allergiesErrorText}
            isError={isError}
            width={'320px'}
          />
          <div style={{ height: '30px' }}>
            <Image
              style={{ cursor: 'pointer' }}
              preview={false}
              width={12}
              alt="allergies-add-icon"
              height={12}
              src={AddIcon}
              onClick={() =>
                props?.caseDetails?.caseName !== ''
                  ? props.handleAddCondition(props.parentIndex)
                  : handleError()
              }
            />
          </div>
        </div>
      )}

      {props.caseDetails?.child?.map((c, index) => (
        <div
          key={`${index}_${c.condition}_${c.operation}`}
          id="allergies-addcondition"
        >
          {props.isChild ? (
            <div
              className={
                c.condition === '' ? 'vertical-line-condition' : 'vertical-line'
              }
            />
          ) : (
            <div
              style={{
                borderLeft: `2px ${
                  c.condition === '' ? 'dashed' : 'solid'
                } #D3D6D9`,
                height: '32px',
              }}
            />
          )}
          {c.condition === '' ? (
            <AddCondition
              caseDetails={c}
              isChild={props.isChild}
              parentId={props.caseDetails.id}
              parentIndex={props.parentIndex}
              value={c.operation}
              addConditionOptions={AllergiesOptions}
              handleAddCondition={props.handleAddCondition}
              handleChangeCaseName={props.handleChangeCaseName}
              conditionName="allergies"
              getPopupContainer={'allergies-addcondition'}
            />
          ) : c.condition === 'reactions' ? (
            <Reactions
              reactions={Symptomarray}
              caseDetails={c}
              isChild={props.isChild}
              parentIndex={props.parentIndex}
              parentId={props.caseDetails.id}
              operationValue={c.operation}
              totalpatient={c?.count}
              handleAddCondition={props.handleAddCondition}
              handleChangeCaseName={props.handleChangeCaseName}
              getPopupContainer="indication"
            />
          ) : (
            ''
          )}
        </div>
      ))}
    </div>
  );
};

Allergies.propTypes = {
  caseDetails: PropTypes.array,
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Allergies.defaultProps = {
  caseDetails: [],
  parentIndex: 0,
  isChild: false,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};
export default Allergies;
