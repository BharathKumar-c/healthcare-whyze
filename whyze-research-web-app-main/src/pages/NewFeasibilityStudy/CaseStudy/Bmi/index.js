import React from 'react';
import PropTypes from 'prop-types';
import { Select, Text } from '../../../../components/basic';
import { bmiRangeOptions } from '../../../../constant';
import { ConditionContainer } from '../../../../components/shared';
import { Row, Col, Tag } from 'antd';
import { Drag, Vector } from '../../../../assets';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { setSegments } from '../../../../redux/reducer/feasibilityStudyReducer';
import { totalPatientFormatter } from '../../../../utils/calculations';
import useNewFeasibilityStudyService from '../../useNewFeasibilityStudyService';

const Bmi = (props) => {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();
  const { optimusValue } = useNewFeasibilityStudyService();

  const handleSelect = (range) => {
    const study = JSON.parse(JSON.stringify(segments));
    study[activeSegment.index].feasibilityStudy[
      props.parentIndex
    ].child[0].bmiRange = range;

    optimusValue({
      id: props.caseDetails.child[0].id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });

    dispatch(setSegments(study));
  };

  return (
    <div className="bmi_container">
      <ConditionContainer className={'condition-container'} isActive isVertical>
        <Row gutter={{ sm: 32, md: 32, lg: 32 }} className="bmi-wrapper">
          <Col className="bmi_drag-icon">
            <Drag />
          </Col>
          <Col>
            <Text className="bmi-wrapper-title" text="BMI Range" />
          </Col>
          <Col>
            <div
              className="condition-container-select"
              id="condition-container-select"
            >
              <Select
                lable="Select type"
                options={bmiRangeOptions}
                getPopupContainer={() =>
                  document.getElementById('condition-container-select')
                }
                name="range"
                value={props.caseDetails?.child[0]?.bmiRange}
                handleSelect={handleSelect}
              />
            </div>
          </Col>
        </Row>
        <Row className="indication-no-patient-tag-row">
          <Tag className="indication-no-patient-tag">
            <Vector />
            {totalPatientFormatter(props.caseDetails?.child[0]?.count || 0)}
          </Tag>
        </Row>
      </ConditionContainer>
    </div>
  );
};

Bmi.propTypes = {
  caseDetails: PropTypes.object,
  parentIndex: PropTypes.number,
};

Bmi.defaultProps = {
  caseDetails: null,
  parentIndex: 0,
};

export default Bmi;
