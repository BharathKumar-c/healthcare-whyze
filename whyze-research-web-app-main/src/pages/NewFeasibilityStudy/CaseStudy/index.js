import React, { useState, useEffect } from 'react';
import { Card } from '../../../components/hoc';
import { useDispatch, useSelector } from 'react-redux';
import { Space } from 'antd';
import Indication from './Indication';
import {
  setCardDetails,
  setSegments,
} from '../../../redux/reducer/feasibilityStudyReducer';
import { v4 as uuid } from 'uuid';
import PropTypes from 'prop-types';
import Medication from './Medication';
import Nicotine from './Nicotine';
import Symptom from './Symptom';
import Bmi from './Bmi';
import Demographics from './Demographics';
import Allergies from './Allergies';
import Surgery from './Surgery';
import Weight from './Weight';
import NoOfPatient from '../../../components/hoc/card/NoOfPatient';
import { cardBorderColors } from '../../../constant';
import useNewFeasibilityStudyService from '../useNewFeasibilityStudyService';
import { getPatientCount } from '../../../utils/Formatter';

export default function CaseStudy() {
  const dispatch = useDispatch();
  const [renameIndex, setRenameIndex] = useState();
  const { cardValue, segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const { optimusValue } = useNewFeasibilityStudyService();

  useEffect(() => {
    const root = document.documentElement;
    root?.style.setProperty(
      '--card-border-color',
      cardBorderColors[cardValue.case],
    );
  }, [cardValue]);

  const handleAddCondition = (index) => {
    const study = JSON.parse(JSON.stringify(segments));
    study[activeSegment.index].feasibilityStudy[index].child = [
      ...study[activeSegment.index].feasibilityStudy[index].child,
      {
        id: uuid().slice(0, 8),
        condition: '',
        operation: 'And',
        child: [],
      },
    ];
    dispatch(setSegments(study));
  };

  const handleChangeCaseName = (data, index) => {
    const study = JSON.parse(JSON.stringify(segments));
    if (typeof data === 'string') {
      study[activeSegment.index].feasibilityStudy[index].caseName = data;
    } else {
      study[activeSegment.index].feasibilityStudy[index].caseName =
        data?.target?.value;
    }

    dispatch(setSegments(study));
  };

  const handleRename = (name) => {
    let newSegments = JSON.parse(JSON.stringify(segments));
    newSegments[activeSegment.index].feasibilityStudy[renameIndex] = {
      ...newSegments[activeSegment.index].feasibilityStudy[renameIndex],
      name,
    };
    dispatch(setSegments(newSegments));
    setRenameIndex();
  };

  const handleSelectOption = (option, index) => {
    if (option === 'Delete') {
      let newSegments = JSON.parse(JSON.stringify(segments));
      newSegments[activeSegment.index].feasibilityStudy.splice(index, 1);

      const feasibilityStudy =
        newSegments[activeSegment.index].feasibilityStudy;
      const cardDetails = {
        case: 'Condition',
        condition: '',
        index: null,
      };

      if (index > 0 && feasibilityStudy[index - 1]) {
        cardDetails.case = feasibilityStudy[index - 1].case;
        cardDetails.condition = feasibilityStudy[index - 1].condition;
        cardDetails.index = index - 1;
      } else if (feasibilityStudy[index]) {
        cardDetails.case = feasibilityStudy[index].case;
        cardDetails.condition = feasibilityStudy[index].condition;
        cardDetails.index = index;
      }
      setRenameIndex();
      dispatch(setCardDetails(cardDetails));
      dispatch(setSegments(newSegments));

      optimusValue({
        name: cardDetails.index !== null ? feasibilityStudy[index]?.name : '',
        data: newSegments[activeSegment.index].feasibilityStudy,
      });
    }

    if (option === 'Duplicate') {
      const newSegments = JSON.parse(JSON.stringify(segments));
      newSegments[activeSegment.index].feasibilityStudy = [
        ...newSegments[activeSegment.index].feasibilityStudy,
        segments[activeSegment.index].feasibilityStudy[index],
      ];
      dispatch(setSegments(newSegments));
      setRenameIndex();
    }

    if (option === 'Rename') {
      setRenameIndex(index);
    }
  };

  const handleClickCard = (index, caseDetails) => {
    dispatch(
      setCardDetails({
        index: index,
        case: caseDetails.case,
        condition: caseDetails.condition,
      }),
    );
  };

  return (
    <div id="casestudy-container">
      {segments[activeSegment.index]?.feasibilityStudy?.map(
        (caseDetails, index) => {
          return (
            <Card
              key={`${index}_${caseDetails.case}`}
              index={index}
              isRename={index === renameIndex}
              title={{ case: caseDetails.case, name: caseDetails.name }}
              tagtitle={caseDetails.condition}
              handleActionClick={() =>
                index === cardValue.index
                  ? null
                  : handleClickCard(index, caseDetails)
              }
              handleSelectOption={handleSelectOption}
              className={
                cardValue.index === index ? 'card-head' : 'inActive-card-head'
              }
              handleRename={handleRename}
              extra={
                <NoOfPatient
                  totalpatient={getPatientCount(caseDetails)}
                  handleSelectOption={(option) =>
                    handleSelectOption(option, index)
                  }
                  getPopupContainer={'casestudy-container'}
                />
              }
            >
              <Space direction="vertical" className="case-study-container">
                {caseDetails.case === 'Demographics' && (
                  <Demographics
                    currentIndex={index}
                    caseDetails={caseDetails.child}
                    type={'number'}
                    label={'Age range'}
                  />
                )}
                {caseDetails.case === 'Indication' && (
                  <Indication
                    parentIndex={index}
                    caseDetails={caseDetails}
                    handleAddCondition={handleAddCondition}
                    handleChangeCaseName={handleChangeCaseName}
                  />
                )}

                {caseDetails.case === 'Medication' && (
                  <Medication
                    parentIndex={index}
                    caseDetails={caseDetails}
                    handleAddCondition={handleAddCondition}
                    handleChangeCaseName={handleChangeCaseName}
                  />
                )}
                {caseDetails.case === 'Nicotine' && (
                  <Nicotine
                    parentIndex={index}
                    caseDetails={caseDetails}
                    handleAddCondition={handleAddCondition}
                    handleChangeCaseName={handleChangeCaseName}
                  />
                )}
                {caseDetails.case === 'BMI' && (
                  <Bmi parentIndex={index} caseDetails={caseDetails} />
                )}
                {caseDetails.case === 'Symptom' && <Symptom />}
                {caseDetails.case === 'Allergies' && (
                  <Allergies
                    parentIndex={index}
                    caseDetails={caseDetails}
                    handleAddCondition={handleAddCondition}
                    handleChangeCaseName={handleChangeCaseName}
                  />
                )}

                {caseDetails.case === 'Weight' && (
                  <Weight
                    parentIndex={index}
                    caseDetails={caseDetails.child}
                    handleAddCondition={handleAddCondition}
                    handleChangeCaseName={handleChangeCaseName}
                  />
                )}

                {caseDetails.case === 'Surgery' && (
                  <Surgery
                    parentIndex={index}
                    caseDetails={caseDetails}
                    handleAddCondition={handleAddCondition}
                    handleChangeCaseName={handleChangeCaseName}
                  />
                )}
              </Space>
            </Card>
          );
        },
      )}
    </div>
  );
}

CaseStudy.propTypes = {
  handleCaseData: PropTypes.func,
};

CaseStudy.defaultProps = {
  handleCaseData: () => {},
};
