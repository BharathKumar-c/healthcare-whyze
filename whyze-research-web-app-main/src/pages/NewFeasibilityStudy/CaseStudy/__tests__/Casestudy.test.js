import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../redux/store';
import React from 'react';
import CaseStudy from '..';

function renderWithContext(element) {
  const { container } = render(<Provider store={store}>{element}</Provider>);
  return { container, store };
}

describe('create snapshot for CaseStudy', () => {
  test('create snapshots', () => {
    const { container } = renderWithContext(<CaseStudy value={2000} />);
    expect(container).toMatchSnapshot();
  });
});
