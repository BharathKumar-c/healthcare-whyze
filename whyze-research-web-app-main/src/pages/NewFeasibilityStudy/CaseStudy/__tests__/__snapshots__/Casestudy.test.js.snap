// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot for CaseStudy create snapshots 1`] = `
<div>
  <div>
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-21"
      >
        <div
          class="ant-card ant-card-bordered card-head"
          id="card_hoc"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 20px;"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-right: 20px;"
                      >
                        <div>
                          <svg>
                            Indication.svg
                          </svg>
                        </div>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <div
                          style="font-weight: 700; font-size: 24px;"
                        >
                          Indication
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <span
                      class="ant-tag titlecomponenet-tag"
                    >
                      Inclusion
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="ant-card-extra"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-end"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 20px;"
                  >
                    <div
                      class="card-component__Vector-icon"
                    />
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <svg
                      class="ant-dropdown-trigger"
                      style="cursor: pointer;"
                    >
                      options.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-space ant-space-vertical case-study-container"
            >
              <div
                class="ant-space-item"
              >
                <div
                  class="indication"
                >
                  <div
                    class="autoComplete-wrapper"
                  >
                    <span
                      class="ant-typography input-lable"
                    >
                      Medical condition
                    </span>
                    <div
                      class="ant-select ant-select-auto-complete autoComplete-wrapper-input ant-select-single ant-select-show-search"
                      style="width: 320px;"
                    >
                      <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                            aria-autocomplete="list"
                            aria-controls="rc_select_TEST_OR_SSR_list"
                            aria-haspopup="listbox"
                            aria-owns="rc_select_TEST_OR_SSR_list"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            id="rc_select_TEST_OR_SSR"
                            role="combobox"
                            type="search"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-placeholder"
                        >
                          Medical condition
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    style="height: 30px;"
                  >
                    <div
                      class="ant-image"
                      style="width: 11px; height: 11px;"
                    >
                      <img
                        alt="Indication-add-icon"
                        class="ant-image-img"
                        height="11"
                        src="AddIcon.svg"
                        style="height: 11px; cursor: pointer;"
                        width="11"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-21"
      >
        <div
          class="ant-card ant-card-bordered inActive-card-head"
          id="card_hoc"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 20px;"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-right: 20px;"
                      >
                        <div>
                          <svg>
                            Demographics.svg
                          </svg>
                        </div>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <div
                          style="font-weight: 700; font-size: 24px;"
                        >
                          Demographics
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <span
                      class="ant-tag titlecomponenet-tag"
                    >
                      Inclusion
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="ant-card-extra"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-end"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 20px;"
                  >
                    <div
                      class="card-component__Vector-icon"
                    />
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <svg
                      class="ant-dropdown-trigger"
                      style="cursor: pointer;"
                    >
                      options.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-space ant-space-vertical case-study-container"
            >
              <div
                class="ant-space-item"
              >
                <div
                  class="demographics_container"
                >
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center condition-container"
                    style="border: 2px solid #d3d6d9; display: flex; flex-direction: column; margin-bottom: 0.5rem; padding: 20px 30px 12px 20px;"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 8px;"
                    >
                      <div
                        class="ant-row"
                      >
                        <h4>
                          Sex
                        </h4>
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                      style="margin-right: 8px;"
                    >
                      <div
                        class="ant-row demographics_gender-wrapper"
                        style="margin-left: -16px; margin-right: -16px;"
                      >
                        <div
                          class="ant-col demographics_gender-wrapper_drag-icon ant-col-md-4 ant-col-lg-4 ant-col-xl-2"
                          style="padding-left: 16px; padding-right: 16px;"
                        >
                          <svg>
                            Drag.svg
                          </svg>
                        </div>
                        <div
                          class="ant-col demographics_gender-wrapper_checkbox ant-col-sm-19 ant-col-md-17 ant-col-lg-14 ant-col-xl-14"
                          style="padding-left: 16px; padding-right: 16px;"
                        >
                          <div
                            class="input-group"
                          >
                            <label
                              class="ant-checkbox-wrapper"
                            >
                              <span
                                class="ant-checkbox"
                              >
                                <input
                                  class="ant-checkbox-input"
                                  name="Male"
                                  type="checkbox"
                                  value=""
                                />
                                <span
                                  class="ant-checkbox-inner"
                                />
                              </span>
                            </label>
                            <span
                              class="ant-typography"
                              style="margin-left: 1rem;"
                            >
                              Male
                            </span>
                          </div>
                          <div
                            class="input-group"
                          >
                            <label
                              class="ant-checkbox-wrapper"
                            >
                              <span
                                class="ant-checkbox"
                              >
                                <input
                                  class="ant-checkbox-input"
                                  name="Female"
                                  type="checkbox"
                                  value=""
                                />
                                <span
                                  class="ant-checkbox-inner"
                                />
                              </span>
                            </label>
                            <span
                              class="ant-typography"
                              style="margin-left: 1rem;"
                            >
                              Female
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <div
                        class="ant-row demographics_gender-wrapper_selected-noof-patients"
                      >
                        <div
                          class="ant-col"
                        >
                          <span
                            class="ant-tag demographics_gender-wrapper_totalpatient-tag"
                          >
                            <svg>
                              Vector-2.svg
                            </svg>
                            <span>
                               
                              0
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center condition-container"
                    style="border: 2px solid #d3d6d9; display: flex; flex-direction: column; margin-bottom: 0.5rem; padding: 20px 30px 12px 20px;"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 8px;"
                    >
                      <div
                        class="ant-row demographics_agerange-wrapper"
                        style="margin-left: -16px; margin-right: -16px;"
                      >
                        <div
                          class="ant-col demographics_agerange-wrapper_drag-icon ant-col-sm-1 ant-col-md-2 ant-col-lg-1 ant-col-xl-1"
                          style="padding-left: 16px; padding-right: 16px;"
                        >
                          <svg>
                            Drag.svg
                          </svg>
                        </div>
                        <div
                          class="ant-col ant-col-sm-19 ant-col-md-17 ant-col-lg-14 ant-col-xl-14 ant-col-xxl-14"
                          style="padding-left: 16px; padding-right: 16px;"
                        >
                          <div
                            class="ant-row"
                          >
                            <div
                              class="ant-col demographics_agerange-wrapper_Input-box ant-col-sm-8 ant-col-md-10 ant-col-lg-9 ant-col-xl-6 ant-col-xxl-3"
                            >
                              <div>
                                <span
                                  class="ant-typography input-lable"
                                >
                                  Age range
                                </span>
                                <input
                                  class="ant-input"
                                  name="min"
                                  placeholder=""
                                  style="border-radius: 0px; width: 100px;"
                                  type="number"
                                  value="0"
                                />
                              </div>
                            </div>
                            <div
                              class="ant-col demographics_agerange-wrapper_Input-box1"
                            >
                              <div>
                                <input
                                  class="ant-input"
                                  name="max"
                                  placeholder=""
                                  style="border-radius: 0px; width: 100px;"
                                  type="number"
                                  value="0"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <div
                        class="ant-row demographics_agerange-wrapper_selected-noof-patients"
                      >
                        <div
                          class="ant-col"
                        >
                          <span
                            class="ant-tag demographics_agerange-wrapper_totalpatient-tag"
                          >
                            <svg>
                              Vector-2.svg
                            </svg>
                            <span>
                               
                              0
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
